### ISSUE TITLE
SPIR-V intrinsics spirv_execution_mode_id not accepting specialization constants

### ISSUE THREAD

**Issue by yavn** [KhronosGroup/glslang#2774]
I noticed an error with using a specialization constant with execution mode intrinsic that takes an id:

```glsl
layout(constant_id = 0) const uint MyConst = 42;
spirv_execution_mode_id(1, MyConst);  // mode 1 is not important here, just any value would do
```

This results in an error:
> Assertion failed: extraOperand != nullptr, file glslang\\MachineIndependent\\SpirvIntrinsics.cpp, line 137

If a regular constant is being used, it compiles correctly:

```glsl
const uint MyConst = 42;
spirv_execution_mode_id(1, MyConst);
```

I suppose this is a bug and a specialization constant should work as well.

**Comment by yavn** [comment]
FYI: @amdrexu

**Comment by greg-lunarg** [comment]
@amdrexu Can you please take this or pass a name of someone who can?

**Comment by amdrexu** [comment]
I will check this. Seems a bug.

**Comment by greg-lunarg** [comment]
Fixed by #2777
